cabot/cabotapp/tests/test_web.py

```diff
@@ -83,3 +88,61 @@ def test_services_list(self):
         # check the extra fields are correct
         self.assertEquals(service.active_checks_count, 3)
         self.assertEquals(service.inactive_checks_count, 1)
+
+
+class TestWebConcurrency(APITransactionTestCase):
+    def setUp(self):
+        self.http_check = HttpStatusCheck.objects.create(
+            id=10102,
+            name='Http Check',
+            importance=Service.CRITICAL_STATUS,
+            endpoint='http://arachnys.com',
+            timeout=10,
+            status_code='200',
+            text_match=None,
+        )
+        self.service = Service.objects.create(
+            id=2194, name='Service',
+        )
+        self.service.save()
+        self.service.status_checks.add(self.http_check)
+
+        self.client = Client()
+
+    def test_delete_service(self):
```

Comment:
this would also pass if the delete just finishes before we try updating right? so it's not really exactly testing the functionality?

Reply:
Yeah, but I can't think of a better way to test for a concurrency issue like this. Ideally I'd put a sleep between when `service.delete()` starts cleaning up related tables and when it deletes the service row, but I can't really do that (without some really gross django-version-dependant mocks).